impress-org/givewp-next-gen
# Feature: Add terms and conditions block

## Description

This PR includes a Terms and conditions block. The block is restricted for a single use and the data is passed through a new "Consent" field for the field api. The terms and conditions block allows users to display their terms and conditions through 3 different display types when selecting the "Show terms" button on the block.

The defaults for the block come from the Global settings but users can customize this per form by selecting the customize option which will enable additional block controls.


Display styles include the following :

### 1) Show terms in modal
Renders the terms in a modal pop over. Agreeing to the terms Sets the checkbox to a checked state. Selecting this option will enable 3 additional block controls for editing the `agreement text`, `Modal Header`& `Modal accept` button.

### 2) Show terms in form
Renders the terms inside the form just beneath the checkbox. Selecting this option will enable an additional block control for editing the `agreement text`.

### 3) Show terms in link
Opens a new window and navigates users to the URL that holds the terms. Selecting this option will enable an additional block control for editing the `URL`.

## Affects

Form builder 
## Visuals

https://www.loom.com/share/39d0cf64c17943339d022dad9a4efee1?sid=29b118e7-c086-4fd3-b0b0-cd0f0bca5bcc

## Testing Instructions

- Add terms and conditions block to a new v3 form. 
- Update all block controls for each display type. 
- Verify the updated changes reflect on Design preview.
- Verify the updated changes reflect on the form page.

Watch loom video for additional clarification.

## Pre-review Checklist






---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1205012554464778
